pkg/server/svid/observer_test.go

```diff
@@ -0,0 +1,14 @@
+package svid
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestObserverFunc(t *testing.T) {
+	var f ObserverFunc = func() State {
+		return State{}
```

Comment:
It might be worth putting something unique in the state to ensure that the observer func returns the exact state (as opposed to returning a zero-value state).